nxc/protocols/ssh.py

```diff
@@ -111,76 +209,82 @@ def plaintext_login(self, username, password, private_key=None):
                     cred_id = self.db.add_credential(
                         "key",
                         username,
-                        password if password != "" else "",
+                        "",
                         key=key_data,
                     )
             else:
-                self.logger.debug(f"Logging in with password")
-                self.conn.connect(
-                    self.host,
-                    port=self.args.port,
-                    username=username,
-                    password=password,
-                    look_for_keys=False,
-                    allow_agent=False,
-                )
-                cred_id = self.db.add_credential("plaintext", username, password)
+                self.logger.debug(f"Logging {self.host} with username: {self.username}, password: {self.password}")
+                self.conn._transport.auth_password(username, password, fallback=True)
 
+            # Some IOT devices will not raise exception in self.conn._transport.auth_password / self.conn._transport.auth_publickey
+            stdin, stdout, stderr = self.conn.exec_command("id")
+            stdout = stdout.read().decode("utf-8", errors="ignore")
+        except Exception as e:
+            self.logger.fail(f"{username}:{process_secret(password) if not pkey else password} {e}")
+            self.conn.close()
+            return False
+        else:
             shell_access = False
             host_id = self.db.get_hosts(self.host)[0].id
 
-            if self.check_if_admin():
-                shell_access = True
-                self.logger.debug(f"User {username} logged in successfully and is root!")
-                if self.args.key_file:
-                    self.db.add_admin_user("key", username, password, host_id=host_id, cred_id=cred_id)
+            if not stdout:
+                stdin, stdout, stderr = self.conn.exec_command("whoami /priv")
+                stdout = stdout.read().decode("utf-8", errors="ignore")
+                self.server_os_platform = "Windows"
+                self.user_principal = "admin"
+                if "SeDebugPrivilege" in stdout:
+                    self.admin_privs = True
+                elif "SeUndockPrivilege" in stdout:
+                    self.admin_privs = True
+                    self.user_principal = "admin (UAC)"
                 else:
-                    self.db.add_admin_user(
-                        "plaintext",
-                        username,
-                        password,
-                        host_id=host_id,
-                        cred_id=cred_id,
-                    )
+                    # non admin (low priv)
+                    self.user_principal = "admin (low priv)"
```

Comment:
Although this is working out in the end with the "non" in line 271 if `self.admin_privs` is false, only seeing these two lines is really confusing :D

I would prefer adding the "non admin" here in this line rather than 271 if that is okay :)

Reply:
Fixed